gddo-server/client.go

```diff
@@ -47,32 +48,15 @@ func timeoutDial(network, addr string) (net.Conn, error) {
 	return timeoutConn{c}, nil
 }
 
-type transport struct {
-	t http.Transport
-}
-
-func (t *transport) RoundTrip(req *http.Request) (*http.Response, error) {
-	timer := time.AfterFunc(*requestTimeout, func() {
-		t.t.CancelRequest(req)
-		log.Printf("Canceled request for %s", req.URL)
-	})
-	defer timer.Stop()
-	if req.URL.Host == "api.github.com" && gitHubCredentials != "" {
-		if req.URL.RawQuery == "" {
-			req.URL.RawQuery = gitHubCredentials
-		} else {
-			req.URL.RawQuery += "&" + gitHubCredentials
-		}
-	}
-	if userAgent != "" {
-		req.Header.Set("User-Agent", userAgent)
-	}
-	return t.t.RoundTrip(req)
-}
-
-var httpClient = &http.Client{Transport: &transport{
-	t: http.Transport{
+var transport = httputil.NewTransportFromEnvironment(
+	&http.Transport{
 		Proxy: http.ProxyFromEnvironment,
 		Dial:  timeoutDial,
 		ResponseHeaderTimeout: *requestTimeout / 2,
-	}}}
+	},
+)
+
+var httpClient = &http.Client{
+	Transport: transport,
+	Timeout:   *requestTimeout,
```

Comment:
This will use the value that requestTimeout points to at init time, before flags are parsed. Probably not what you want.